cpp/include/raft/random/detail/rng_impl.cuh

```diff
@@ -37,10 +32,99 @@ namespace detail {
 enum GeneratorType {
   /** curand-based philox generator */
   GenPhilox = 0,
-  /** LFSR taps generator */
-  GenTaps,
-  /** kiss99 generator (currently the fastest) */
-  GenKiss99
+  /** Permuted Congruential Generator */
+  GenPC
+};
+
+template <typename OutType>
```

Comment:
Since this is c++ code, we might want to consider adding some constructors to each of these to make it easier for users to construct these. Also, I don't believe the documentation propagates through the `using` clause, unfortunately, so it might be worth exposing these through an `rng_common.hpp` header which can be included both in the `detail` and public namespaces.